src/res/mod.rs

```diff
@@ -36,68 +45,53 @@ where
     }
 }
 
-impl<'a, T> SystemData<'a> for Fetch<'a, T>
+impl<'a, T, F> SystemData<'a> for Fetch<'a, T, F>
 where
     T: Resource,
+    F: FallbackHandler<T>,
 {
     fn fetch(res: &'a Resources) -> Self {
-        res.fetch()
+        F::fetch(res)
```

Comment:
Why does `FallbackHandler`  does fetching instead of being touched only when fallback is required?
This is inconsistent with `SystemData` implementation for `Option<Fetch/FetchMut>` which doesn't use `FallbackHandler` as it doesn't need one,

Reply:
The `FallbackHandler` does the complete fetching since that is better for performance. It can use the `Entry` API to do insertions.

> This is inconsistent with SystemData implementation for Option<Fetch/FetchMut> which doesn't use FallbackHandler as it doesn't need one

I don't see how it is inconsistent. You said it yourself, `Option<Fetch>` does not _need_ a fallback.